drivers/display/display_nt35510.c

```diff
@@ -0,0 +1,417 @@
+	ret += nt35510_write_reg(dev, 0xB6, nt35510_reg18,
+				 1); /* Set source output data hold time */
+	ret += nt35510_write_reg(dev, 0xB5, nt35510_reg19, 1); /* Display resolution control */
+	ret += nt35510_write_reg(dev, 0xB7, nt35510_reg20, 2); /* Gate EQ control */
+	ret += nt35510_write_reg(dev, 0xB8, nt35510_reg21, 4); /* Src EQ control(Mode2) */
+	ret += nt35510_write_reg(dev, 0xBC, nt35510_reg22, 3);
+	ret += nt35510_write_reg(dev, 0xCC, nt35510_reg23, 3);
+	ret += nt35510_write_reg(dev, 0xBA, nt35510_reg24, 1);
+
+	/* Add a delay, otherwise MADCTL not taken */
+	k_msleep(200);
+
+	/* Configure orientation */
+	if (data->orientation == DISPLAY_ORIENTATION_NORMAL) {
+		ret += nt35510_write_reg_val(dev, NT35510_CMD_MADCTL, 0x00);
+		ret += nt35510_write_reg(dev, NT35510_CMD_CASET, nt35510_caset_portrait, 4);
+		ret += nt35510_write_reg(dev, NT35510_CMD_RASET, nt35510_raset_portrait, 4);
+	} else {
+		ret += nt35510_write_reg_val(dev, NT35510_CMD_MADCTL, 0x60);
+		ret += nt35510_write_reg(dev, NT35510_CMD_CASET, nt35510_caset_landscape, 4);
+		ret += nt35510_write_reg(dev, NT35510_CMD_RASET, nt35510_raset_landscape, 4);
+	}
+
+	/* Exit sleep mode */
+	ret += nt35510_write_reg(dev, NT35510_CMD_SLPOUT, NULL, 0);
+
+	/* Wait for sleep out exit */
+	k_msleep(20);
+
+	/* Set color mode */
+	ret += nt35510_write_reg_val(dev, NT35510_CMD_COLMOD,
+				     data->dsi_pixel_format == PIXEL_FORMAT_RGB_565
+					     ? NT35510_COLMOD_RGB565
+					     : NT35510_COLMOD_RGB888);
+
+	/** CABC : Content Adaptive Backlight Control section start >> */
+	/* Note : defaut is 0 (lowest Brightness], 0xFF is highest Brightness, try 0x7F :
+	 * intermediate value */
+	// ret += nt35510_write_reg(dev, NT35510_CMD_WRDISBV, &nt35510_reg31[1], 1);
+	ret += nt35510_write_reg_val(dev, NT35510_CMD_WRDISBV, 0x7f);
+
+	/* defaut is 0, try 0x2C - Brightness Control Block, Display Dimming & BackLight on */
+	// ret += nt35510_write_reg(dev, NT35510_CMD_WRCTRLD, &nt35510_reg32[1], 1);
+	ret += nt35510_write_reg_val(dev, NT35510_CMD_WRCTRLD, 0x2c);
+
+	/* defaut is 0, try 0x02 - image Content based Adaptive Brightness [Still Picture] */
+	// ret += nt35510_write_reg(dev, NT35510_CMD_WRCABC, &nt35510_reg33[1], 1);
+	ret += nt35510_write_reg_val(dev, NT35510_CMD_WRCABC, 0x2);
+
+	/* defaut is 0 (lowest Brightness], 0xFF is highest Brightness */
+	// ret += nt35510_write_reg(dev, NT35510_CMD_WRCABCMB, &nt35510_reg34[1], 1);
+	ret += nt35510_write_reg_val(dev, NT35510_CMD_WRCABCMB, 0xff);
+
+	/* Send Command Display On */
+	ret += nt35510_write_reg(dev, MIPI_DCS_SET_DISPLAY_ON, NULL, 0);
+
+	/* Send Command GRAM memory write (no parameters)
+	   this initiates frame write via other DSI commands sent by
+	   DSI host from LTDC incoming pixels in video mode */
+	ret += nt35510_write_reg(dev, NT35510_CMD_RAMWR, NULL, 0);
+
+	return ret;
+}
+
+static int nt35510_write(const struct device *dev, uint16_t x, uint16_t y,
```

Comment:
These stub functions aren't needed- the display API will return `-ENOSYS` if they are not present

Reply:
Ok, so if i understand you correctly, you mean i can remove
the stub and let write function pointer be initialized to NULL?

Reply:
Yes, exactly. Just remove the function definition and entry in the API structure.